test/modellib/ScanModel.cpp

```diff
@@ -0,0 +1,221 @@
+  return exec != nullptr;
+}
+
+bool ScanLibBuilder::compileAndLoad(const onnx_mlir::CompilerOptionList &list) {
+  if (setCompilerOptions(list) != 0)
+    return false;
+  return compileAndLoad();
+}
+
+bool ScanLibBuilder::prepareInputs() {
+  constexpr int num = 2;
+  OMTensor **list = (OMTensor **)malloc(num * sizeof(OMTensor *));
+  if (!list)
+    return false;
+  list[0] =
+      omTensorCreateWithRandomData<float>(llvm::makeArrayRef(initialShape));
+  list[1] = omTensorCreateWithRandomData<float>(llvm::makeArrayRef(xShape));
+  // Compute reference. Scan with onnx.Add
+  for (int64_t b = 0; b < B; ++b) {
+    for (int64_t i = 0; i < I; ++i) {
+      omTensorGetElem<float>(list[0], {b, i}) = 0;
+    }
+  }
+  int n = 1;
+  for (int64_t b = 0; b < B; ++b) {
+    for (int64_t s = 0; s < S; ++s) {
+      for (int64_t i = 0; i < I; ++i) {
+        omTensorGetElem<float>(list[1], {b, s, i}) = (float) n++;
+      }
+    }
+  }
+  inputs = omTensorListCreateWithOwnership(list, num, true);
+  return inputs && list[0] && list[1];
+}
+
+bool ScanLibBuilder::prepareInputs(float dataRange) {
+  constexpr int num = 2;
+  OMTensor **list = (OMTensor **)malloc(num * sizeof(OMTensor *));
+  if (!list)
+    return false;
+  list[0] = omTensorCreateWithRandomData<float>(
+      llvm::makeArrayRef(initialShape), -dataRange, dataRange);
+  list[1] = omTensorCreateWithRandomData<float>(
+      llvm::makeArrayRef(xShape), -dataRange, dataRange);
+  inputs = omTensorListCreateWithOwnership(list, num, true);
+  return inputs && list[0] && list[1];
+}
+
+bool ScanLibBuilder::verifyOutputs() {
+  // Get inputs and outputs.
+  if (!inputs || !outputs)
+    return false;
+  OMTensor *init = omTensorListGetOmtByIndex(inputs, 0);
+  OMTensor *x = omTensorListGetOmtByIndex(inputs, 1);
+  OMTensor *resy = omTensorListGetOmtByIndex(outputs, 0);
+  OMTensor *resz = omTensorListGetOmtByIndex(outputs, 1);
+  OMTensor *refy = omTensorCreateWithShape<float>({I});
+  OMTensor *refz = omTensorCreateWithShape<float>({S, I});
+  if (!init || !x || !resy || !refy || !resz || !refz)
+    return false;
+#if 0
+  for (int64_t b = 0; b < B; ++b) {
+    for (int64_t i = 0; i < I; ++i) {
+      printf("init<b=%ld, i=%ld>: %f\n", b, i,
+          omTensorGetElem<float>(init, {b, i}));
+    }
+  }
+  for (int64_t b = 0; b < B; ++b) {
+    for (int64_t s = 0; s < S; ++s) {
+      for (int64_t i = 0; i < I; ++i) {
+        printf("x<b=%ld, s=%ld, i=%ld>: %f\n", b, s, i,
+            omTensorGetElem<float>(x, {b, s, i}));
+      }
+    }
+  }
+#endif
+  // Compute reference. Scan with onnx.Add
+  for (int64_t i = 0; i < I; ++i) {
+    float refVal = 0.0;
+    for (int64_t b = 0; b < B; ++b) {
+      refVal = omTensorGetElem<float>(init, {b, i});
+      for (int64_t s = 0; s < S; s++) {
+        refVal += omTensorGetElem<float>(x, {b, s, i});
+        omTensorGetElem<float>(refz, {s, i}) = refVal;
+      }
+    }
+    omTensorGetElem<float>(refy, {i}) = refVal;
+  }
+#if 0
```

Comment:
please remove in final version, or add a #define with a variable name that can set for debugging.